advanced/Scripts/utils.sh

```diff
@@ -15,46 +17,74 @@
 #  - New functions must have a test added for them in test/test_any_utils.py
 
 #######################
-# Takes three arguments key, value, and file.
+# Takes either
+#   - Three arguments: file, key, and value.
+#   - Two arguments: file, and key.
+#
 # Checks the target file for the existence of the key
 #   - If it exists, it changes the value
 #   - If it does not exist, it adds the value
 #
 # Example usage:
-# addOrEditKeyValuePair "BLOCKING_ENABLED" "true" "/etc/pihole/setupVars.conf"
+# addOrEditKeyValuePair "/etc/pihole/setupVars.conf" "BLOCKING_ENABLED" "true"
 #######################
 addOrEditKeyValPair() {
-  local key="${1}"
-  local value="${2}"
-  local file="${3}"
-  if grep -q "^${key}=" "${file}"; then
-    sed -i "/^${key}=/c\\${key}=${value}" "${file}"
+  local file="${1}"
+  local key="${2}"
+  local value="${3}"
+
+  if [ "${value}" != "" ]; then
+    # value has a value, so it is a key pair
+    if grep -q "^${key}=" "${file}"; then
+      # Key already exists in file, modify the value
+      sed -i "/^${key}=/c\\${key}=${value}" "${file}"
+    else
+      # Key does not already exist, add it and it's value
+      echo "${key}=${value}" >> "${file}"
+    fi
   else
-    echo "${key}=${value}" >> "${file}"
+    # value has no value, so it is just a key. Add it if it does not already exist
+    if ! grep -q "^${key}" "${file}"; then
+      # Key does not exist, add it.
+      echo "${key}" >> "${file}"
+    fi
   fi
 }
 
+#######################
+# Takes two arguments key, and file.
+# Deletes a key from target file
+#
+# Example usage:
+# removeKey "PIHOLE_DNS_1" "/etc/pihole/setupVars.conf"
+#######################
+removeKey() {
```

Comment:
Is it cheeky to ask to change the order of arguments here as well? Just to have the same as in `addOrEditKeyValPair`: file first, key second?